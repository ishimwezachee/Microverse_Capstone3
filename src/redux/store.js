import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import reducer from './action-reducer';

const reducers = combineReducers({
  reducer,
});

const store = createStore(reducers, applyMiddleware(logger, thunk));

export default store;
