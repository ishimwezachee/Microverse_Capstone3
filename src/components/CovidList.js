import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getCovidDataFromApi } from '../redux/action-reducer';
import CovidItem from './CovidItem';

const CovidList = () => {
  const [value, setValue] = useState('');
  const dispatch = useDispatch();
  const data = useSelector((state) => state.reducer);
  const inputHandler = (e) => {
    setValue(e.target.value);
  };

  useEffect(() => {
    dispatch(getCovidDataFromApi());
  }, []);

  return (
    <div className="wraper">
      <input placeholder="Search by country Name" className="input" type="text" value={value} onChange={inputHandler} />
      <br />
      <h2>Covid Details</h2>
      <div className="data">
        {' '}

        {
        value ? (
          data.covid
            .filter((countryData) => countryData[0]
              .toLowerCase()
              .includes(value.toLocaleLowerCase()))
            .map((country) => (
              <CovidItem key={country[0]} country={country[0]} data={country[1]} />
            ))
        )
          : (
            data.covid.map((country) => (
              <CovidItem key={country[0]} country={country[0]} data={country[1]} />
            ))
          )
      }
      </div>
    </div>
  );
};

export default CovidList;
