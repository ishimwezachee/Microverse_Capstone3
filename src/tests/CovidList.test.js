import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import CovidList from '../components/CovidList';

describe('Country Component test', () => {
  it('Renders Deatils', () => {
    const app = renderer.create(
      <Provider store={store}>
        <CovidList />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});
