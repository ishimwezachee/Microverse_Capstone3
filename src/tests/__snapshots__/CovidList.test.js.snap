// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Component test Renders Deatils 1`] = `
<div
  className="wraper"
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Search by country Name"
    type="text"
    value=""
  />
  <br />
  <div
    className="data"
  >
     
    <h1>
      List of countries
    </h1>
  </div>
</div>
`;
